qcodes/instrument/group_parameter.py

```diff
@@ -176,6 +181,26 @@ def __init__(self,
         else:
             self.get_parser = self._separator_parser(separator)
 
+        have_initial_values = [p._initial_value is not None
+                               for p in parameters]
+        if any(have_initial_values):
+            if not all(have_initial_values):
+                have_initial_values = [p.name for p in parameters
```

Comment:
could you please not re-use an existing variable? i suggest `params_with_initial_values` for example.